lib/puzzles/widgets/puzzle_listener.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_and_friends/puzzles/cubit/puzzles_cubit.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PuzzleListener extends StatelessWidget {
+  const PuzzleListener({required this.child, super.key});
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiBlocListener(
```

Comment:
Can we remove the `MultiBlocListener` since there's only a single listener being used here?